packages/app-form-builder/src/admin/components/FormEditor/Tabs/EditTab/FormStep.tsx

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+import { FbFormModelField, FbFormStep } from "~/types";
+import EditFieldDialog from "./EditFieldDialog";
+import { Wrapper, StyledAccordion, StyledAccordionItem, RowHandle } from "./Styled";
+import { Icon } from "@webiny/ui/Icon";
+import { AccordionItem } from "@webiny/ui/Accordion";
+import { ReactComponent as DeleteIcon } from "@material-design-icons/svg/outlined/delete_outline.svg";
+import { ReactComponent as EditIcon } from "@material-design-icons/svg/outlined/edit.svg";
+import { ReactComponent as HandleIcon } from "~/admin/components/FormEditor/icons/round-drag_indicator-24px.svg";
+
+import { EmptyFormStep } from "./FormStep/EmptyFormStep";
+import { FormStepWithFields } from "./FormStep/FormStepWithFields";
+import { useFormStep } from "./FormStep/useFormStep";
+
+export interface FormStepProps {
+    title: string;
+    deleteStepDisabled: boolean;
+    formStep: FbFormStep;
+    onDelete: () => void;
+    onEdit: () => void;
+    getStepFields: (stepId: string) => FbFormModelField[][];
+    updateField: (field: FbFormModelField) => void;
+    deleteField: (field: FbFormModelField, stepId: string) => void;
+}
+
+export const FormStep: React.FC<FormStepProps> = props => {
+    const { title, deleteStepDisabled, formStep, onDelete, onEdit, getStepFields, updateField } =
+        props;
+
+    const { dropDestination, editingField, editField, createCustomField } = useFormStep();
+
+    const fields = getStepFields(formStep.id);
+
+    return (
```

Comment:
Check how I refactored this file. Some of the things I did:

1. split into multiple files, just so there's not a ton of code nesting, with code being broken into multiple lines in weird ways, which makes it hard to read.
2. Memoized things like beginDragParams, used `useCallback` with `onDrop` handlers
3. While doing 2, I used descriptive names for variables. This helps developer figure out what exactly the thing is doing / is responsible for.
4. This whole `FormStep.tsx` now has its own `useFormStep` hook, which contains some of the functionality needed in FormStep.tsx

There might be more.

All in all, personally I like to do refactors in separate PRs, but in this case, the FormStep was just horrible, and wanted to do it ASAP, before we merge this.

Reply:
@adrians5j 
I can see that we use `useFormStep` hook that has a local state in different components. 
So in `FormStep` component and `FormStepRowField` component `editingField` and `dropDestination` state have different values. 
Because of that we have issues with creating new field and editing the existing field. 
So to make things work properly we need to have a shared data between `FormStep` and `FormStepRowField` components in order to create new custom field and edit an existing field. 
Maybe it would be better to create a context for `FormStep` where we could store `editingField` and `dropDestination` values?

Reply:
@adrians5j could you please verify possible solution

Reply:
Hey @neatbyte-ivelychko, yeah sorry about that. That's a mistake on my part.

In any case, I agree, context is the right way to go.

It can be `FormStepContext`, and the name of the hook can still remain the same. At least from what I remember. Would that make sense?

Something like `packages/app-form-builder/src/admin/views/Forms/FormsContext.tsx`?

Reply:
@adrians5j yeah, that makes sense. Thanks